docs/book/developer-guide/advanced-concepts/materializer.md

```diff
@@ -0,0 +1,454 @@
+---
+description: How to use materializers to pass custom data types through steps.
+---
+
+# Materializers
+
+A ZenML pipeline is built in a data-centric way. The outputs and inputs of steps
+define how steps are connected and the order in which they are executed. Each
+step should be considered as its very own process that reads and writes its
+inputs and outputs from and to the
+[Artifact Store](../../mlops_stacks/artifact_stores/overview.md). 
+This is where **Materializers** come into play.
+
```

Comment:
While I totally agree that this would be useful, I would suggest to keep this out of scope for now. 

Personally, I'm not a big fan of materializers, I always felt like this is something that should just be there and work. For those kind of things (same thing getting pipeline runs with post execution) I would rather spend some time brainstorming how to improve the underlying design patterns than creating elaborate figures that show how the current design is working. Also goes a bit into the direction of showing users how to do things vs. showing users how cool the underlying tech is.